### PR TITLE
Full implementation of _fileUpload method

### PR DESCRIPTION
**Pull request by manolo**
connected to #1 

[Review on Reviewable](https://reviewable.io/reviews/vaadin/vaadin-upload/7)

### CODE REVIEW

**Comment by platosha** [comment]
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[test/upload.html, line 35 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/7#-K9b82UywNV4E9yceE5G:-K9b82UzrCwiy0EhGEdz:953654595) ([raw file](https://github.com/vaadin/vaadin-upload/blob/17d4a5a1f5d293ed784648b06ecb11b3738210a4/test/upload.html#L35)):*
Let's rephrase starting with `should` here and onwards. I think we can just remove `on file upload` words, since it was already expressed by grouping them into describe('Upload Events').

---

*[vaadin-upload.html, line 44 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/7#-K9bBSc5zPmMi-n7dK50:-K9bBSc5zPmMi-n7dK51:-703757005) ([raw file](https://github.com/vaadin/vaadin-upload/blob/17d4a5a1f5d293ed784648b06ecb11b3738210a4/vaadin-upload.html#L44)):*
Please consider using events instead of function properties, if possible.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/7)*

**Comment by manolo** [comment]
Review status: 1 of 3 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[test/upload.html, line 35 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/7#-K9b82UywNV4E9yceE5G:-K9bysivdethV1EDsQNr:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/17d4a5a1f5d293ed784648b06ecb11b3738210a4/test/upload.html#L35)):*
Done.

---

*[vaadin-upload.html, line 44 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/7#-K9bBSc5zPmMi-n7dK50:-K9byrPwuUqU2pWqfmcI:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/17d4a5a1f5d293ed784648b06ecb11b3738210a4/vaadin-upload.html#L44)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/7)*

**Comment by platosha** [comment]
Reviewed 3 of 6 files at r3, 2 of 3 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[vaadin-upload.html, line 162 \[r5\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/7#-K9f9p5x0OmFCXLNL3cl:-K9f9p5x0OmFCXLNL3cm:-615209747) ([raw file](https://github.com/vaadin/vaadin-upload/blob/c322a6bb1202186c2c114a54101e8018138b53d0/vaadin-upload.html#L162)):*
I might confused you, but I meant a different thing.

If default is prevented, then we should fire neither fail nor complete. Instead, we should do nothing about the response.

It's a possibility for developers to skip the internal response handling and do something on his own to handle it, like restart uploading, sending another xhr and so on.

In short, just `return` here.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/7)*

**Comment by manolo** [comment]
Review status: 4 of 6 files reviewed at latest revision, 1 unresolved discussion.

---

*[vaadin-upload.html, line 162 \[r5\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/7#-K9f9p5x0OmFCXLNL3cl:-K9fFSLxbSqKCxvrHrSR:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/c322a6bb1202186c2c114a54101e8018138b53d0/vaadin-upload.html#L162)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/7)*

**Comment by platosha** [comment]
Reviewed 1 of 2 files at r6.
Review status: 5 of 6 files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/7)*

**Comment by platosha** [comment]
Reviewed 1 of 2 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/7)*

**Comment by platosha** [comment]
Reviewed 1 of 2 files at r7, 1 of 1 files at r8.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/7)*
